### PR TITLE
FlowDispatcher: Rename to FlowParser

### PR DESCRIPTION
**Pull request by tbarbette**
Not finished, need to extract what is not "parser". FlowManager is taken. So FlowRuleManager? And maybe FlowRuleParser instead of just FlowParser.

### CODE REVIEW

File: configure [hunk 1]
```diff
@@ -11069,15 +11069,15 @@ else
   ac_have_flow_disp_h=no
 fi
 
-as_ac_File=`$as_echo "ac_cv_file_$srcdir/include/click/flowdispatcherparser.hh" | $as_tr_sh`
-{ $as_echo "$as_me:${as_lineno-$LINENO}: checking for $srcdir/include/click/flowdispatcherparser.hh" >&5
-$as_echo_n "checking for $srcdir/include/click/flowdispatcherparser.hh... " >&6; }
+as_ac_File=`$as_echo "ac_cv_file_$srcdir/include/click/flowparserparser.hh" | $as_tr_sh`
```
Comment (gkatsikas) [inline configure:11072]: This is funny :p You could rename it to flowparsersquare :p
This is why FlowRuleManager or NICRuleManager is more relevant. For now , leave it as it is and we will merge properly into master

File: include/click/flowparserglue.hh [hunk 2]
```diff
@@ -20,6 +20,10 @@
 #ifndef CLICK_FLOWDISPATCHER_GLUE_H
```
Comment (gkatsikas) [inline include/click/flowparserglue.hh:20]: These names must be updated accordingly, but let's do it later

File: include/click/flowparserglue.hh [hunk 3]
```diff
@@ -124,6 +128,77 @@ int cmdline_parse(
 
 #ifdef __cplusplus
 }
+
+#define FLOWDISP_ERROR   ((int)-1)
```
Comment (gkatsikas) [inline include/click/flowparserglue.hh:132]: Same here

File: configure (hunk 1 above)
Comment (tbarbette) [reply configure:11072]: That's a mistake actually :p flowparserparser.hh does not exist ^^

**Comment by gkatsikas** [comment]
I suggest to properly address the naming of this component now that there is no time pressure

**Comment by tbarbette** [comment]
Yep! I'd say FlowParser is okay but we need to extract the managing part to FlowRuleManager. We can keep the system name as FlowParser because all in all, all that is for parsing the DPDK format. But that's your code, I'll follow ;)

**Comment by gkatsikas** [comment]
More tomorrow :p

**Comment by gkatsikas** [comment]
I am done with the refactoring, I believe we can merge if Travis agrees

**Comment by gkatsikas** [comment]
> I am done with the refactoring, I believe we can merge if Travis agrees

Actually, there is more to do.
We could decouple the FlowCache into a separate file and rename it to FlowRuleCache.
The remaining part could be renamed to FlowRuleManager

**Comment by gkatsikas** [comment]
Now, the floor is yours :)

**Comment by tbarbette** [comment]
Great! Thanks!

**Pull request merged**